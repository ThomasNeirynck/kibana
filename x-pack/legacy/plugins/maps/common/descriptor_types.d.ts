/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export type EMSXYZSourceType = 'EMS_XYZ';
export type SourceType = EMSXYZSourceType | string;

export interface ISourceDescriptor {
  id: string;
  type: SourceType;
}

export interface IXYZTMSSourceDescriptor extends ISourceDescriptor {
  urlTemplate: string;
  type: EMSXYZSourceType;
}

export interface ILayerDescriptor {
  sourceDescriptor: ISourceDescriptor;
  id: string;
}
