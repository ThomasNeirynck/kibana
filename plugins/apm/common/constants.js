export const APP_NAME = 'context.app.name';
export const APP_AGENT_NAME = 'context.app.agent.name';
export const EVENT_PROCESSOR_NAME = 'processor.name';

export const TRANSACTION_DURATION = 'transaction.duration.us';
export const TRANSACTION_TYPE = 'transaction.type';
export const TRANSACTION_RESULT = 'transaction.result';
export const TRANSACTION_NAME = 'transaction.name';
export const TRANSACTION_ID = 'transaction.id';

export const TRACE_START = 'trace.start.us';
export const TRACE_DURATION = 'trace.duration.us';
export const TRACE_TRANSACTION_ID = 'trace.transaction_id';
export const TRACE_TYPE = 'trace.type';
export const TRACE_NAME = 'trace.name';
export const TRACE_ID = 'trace.id';
export const TRACE_SQL = 'context.db.statement';

export const ERROR_GROUP_ID = 'error.grouping_key';
export const ERROR_CULPRIT = 'error.culprit';
export const ERROR_MESSAGE = 'error.exception.message';
