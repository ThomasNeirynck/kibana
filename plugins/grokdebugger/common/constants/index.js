export { ROUTES } from './routes';
export { PLUGIN } from './plugin';
