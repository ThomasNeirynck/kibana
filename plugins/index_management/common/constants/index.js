export { PLUGIN } from './plugin';
export { BASE_PATH } from './base_path';
export * from './index_statuses';
