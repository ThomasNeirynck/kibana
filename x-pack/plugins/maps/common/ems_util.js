/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */


import { GIS_API_PATH } from './constants';

export  async function getEMSResources(emsClient, includeElasticMapsService, licenseUid, useCors) {

  if (!includeElasticMapsService) {
    return {
      fileLayers: [],
      tmsServices: []
    };
  }

  emsClient.addQueryParams({ license: licenseUid });
  const fileLayerObjs = await emsClient.getFileLayers();
  const tmsServicesObjs = await emsClient.getTMSServices();

  const fileLayers = fileLayerObjs.map(fileLayer => {
    //backfill to static settings
    const format = fileLayer.getDefaultFormatType();
    const meta = fileLayer.getDefaultFormatMeta();

    return {
      name: fileLayer.getDisplayName(),
      origin: fileLayer.getOrigin(),
      id: fileLayer.getId(),
      created_at: fileLayer.getCreatedAt(),
      attribution: fileLayer.getHTMLAttribution(),
      attributions: fileLayer.getAttributions(),
      fields: fileLayer.getFieldsInLanguage(),
      url: useCors ? fileLayer.getDefaultFormatUrl() : `../${GIS_API_PATH}/data/ems?id=${encodeURIComponent(fileLayer.getId())}`,
      format: format, //legacy: format and meta are split up
      meta: meta, //legacy, format and meta are split up,
      emsLink: fileLayer.getEMSHotLink()
    };
  });

  const tmsServices = tmsServicesObjs.map(tmsService => {
    return {
      origin: tmsService.getOrigin(),
      id: tmsService.getId(),
      minZoom: tmsService.getMinZoom(),
      maxZoom: tmsService.getMaxZoom(),
      attribution: tmsService.getHTMLAttribution(),
      attributionMarkdown: tmsService.getMarkdownAttribution(),
      url: tmsService.getUrlTemplate()
    };
  });

  return { fileLayers, tmsServices };
}
