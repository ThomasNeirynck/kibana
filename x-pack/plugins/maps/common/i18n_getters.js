/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';
import { APP_TITLE_EN } from './constants';


export function getAppTitle() {
  return i18n.translate('xpack.maps.appTitle', {
    defaultMessage: APP_TITLE_EN
  });
}
