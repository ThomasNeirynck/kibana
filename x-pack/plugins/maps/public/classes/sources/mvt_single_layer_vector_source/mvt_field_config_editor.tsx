/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
/* eslint-disable @typescript-eslint/consistent-type-definitions */

import React, { ChangeEvent, Component, Fragment } from 'react';
import {
  EuiButton,
  EuiButtonIcon,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSuperSelect,
  EuiFieldText,
  EuiSpacer,
} from '@elastic/eui';
import { i18n } from '@kbn/i18n';
import _ from 'lodash';
import { MVTFieldDescriptor } from '../../../../common/descriptor_types';
import { FieldIcon } from '../../../../../../../src/plugins/kibana_react/public';
import { MVTFieldType } from '../../../../common/constants';

const FIELD_TYPE_OPTIONS = [
  {
    value: MVTFieldType.STRING,
    inputDisplay: (
      <span>
        <FieldIcon type={'string'} />
        <span>
          {i18n.translate('xpack.maps.mvtSource.stringFieldLabel', {
            defaultMessage: 'string',
          })}
        </span>
      </span>
    ),
  },
  {
    value: MVTFieldType.NUMBER,
    inputDisplay: (
      <span>
        <FieldIcon type={'number'} />
        <span>
          {i18n.translate('xpack.maps.mvtSource.numberFieldLabel', {
            defaultMessage: 'number',
          })}
        </span>
      </span>
    ),
  },
];

export interface Props {
  fields: MVTFieldDescriptor[];
  onChange: (fields: MVTFieldDescriptor[]) => void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface State {
  previousFields: MVTFieldDescriptor[];
  currentFields: MVTFieldDescriptor[];
}

export class MVTFieldConfigEditor extends Component<Props, State> {
  state: State = {
    currentFields: [],
    previousFields: [],
  };

  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
    if (_.isEqual(nextProps.fields, prevState.previousFields)) {
      return null;
    }
    const clonedFields = _.cloneDeep(nextProps.fields);
    return {
      currentFields: clonedFields,
      previousFields: clonedFields,
    };
  }

  _notifyChange = _.debounce(() => {
    const invalid = this.state.currentFields.some((field: MVTFieldDescriptor) => {
      return field.name === '';
    });

    if (!invalid) {
      this.props.onChange(this.state.currentFields);
    }
  });

  _fieldChange(newFields: MVTFieldDescriptor[]) {
    this.setState(
      {
        currentFields: newFields,
      },
      this._notifyChange
    );
  }

  _removeField(index: number) {
    const newFields: MVTFieldDescriptor[] = this.state.currentFields.slice();
    newFields.splice(index, 1);
    this._fieldChange(newFields);
  }

  _addField = () => {
    const newFields: MVTFieldDescriptor[] = this.state.currentFields.slice();
    newFields.push({
      type: MVTFieldType.STRING,
      name: '',
    });
    this._fieldChange(newFields);
  };

  _renderFieldTypeDropDown(mvtFieldConfig: MVTFieldDescriptor, index: number) {
    const onChange = (type: MVTFieldType) => {
      const newFields = this.state.currentFields.slice();
      newFields[index] = {
        type,
        name: newFields[index].name,
      };
      this._fieldChange(newFields);
    };

    return (
      <EuiSuperSelect
        options={FIELD_TYPE_OPTIONS}
        valueOfSelected={mvtFieldConfig.type}
        onChange={(value) => onChange(value)}
      />
    );
  }

  _renderFieldNameInput(mvtFieldConfig: MVTFieldDescriptor, index: number) {
    const onChange = (e: ChangeEvent<HTMLInputElement>) => {
      const name = e.target.value;
      const newFields = this.state.currentFields.slice();
      newFields[index] = {
        name,
        type: newFields[index].type,
      };
      this._fieldChange(newFields);
    };

    const emptyName = mvtFieldConfig.name === '';
    let hasDupes = false;
    for (let i = 0; i < this.state.currentFields.length; i++) {
      if (i !== index && mvtFieldConfig.name === this.state.currentFields[i].name) {
        hasDupes = true;
        break;
      }
    }
    const isInvalid = emptyName || hasDupes;
    const placeholderText = isInvalid
      ? i18n.translate('xpack.maps.mvtSource.fieldPlaceholderText', {
          defaultMessage: 'Field name',
        })
      : '';

    return (
      <EuiFieldText
        value={mvtFieldConfig.name}
        onChange={onChange}
        aria-label={'Fieldname'}
        placeholder={placeholderText}
        isInvalid={isInvalid}
      />
    );
  }

  _renderFieldConfig() {
    return this.state.currentFields.map((mvtFieldConfig: MVTFieldDescriptor, index: number) => {
      return (
        <EuiFlexGroup key={index} gutterSize="xs">
          <EuiFlexItem>{this._renderFieldNameInput(mvtFieldConfig, index)}</EuiFlexItem>
          <EuiFlexItem>{this._renderFieldTypeDropDown(mvtFieldConfig, index)}</EuiFlexItem>
          <EuiFlexItem>
            <EuiButtonIcon
              iconType="trash"
              color="danger"
              onClick={() => {
                this._removeField(index);
              }}
              title={i18n.translate('xpack.maps.mvtSource.trashButtonTitle', {
                defaultMessage: 'Remove field',
              })}
              aria-label={i18n.translate('xpack.maps.mvtSource.trashButtonAriaLabel', {
                defaultMessage: 'Remove field',
              })}
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      );
    });
  }

  render() {
    return (
      <Fragment>
        {this._renderFieldConfig()}
        <EuiSpacer size={'xs'} />
        <EuiButton fill isDisabled={false} onClick={this._addField} size="s">
          {'Add field'}
        </EuiButton>
      </Fragment>
    );
  }
}
