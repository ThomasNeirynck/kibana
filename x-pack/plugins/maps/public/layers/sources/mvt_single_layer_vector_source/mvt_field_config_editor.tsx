/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
/* eslint-disable @typescript-eslint/consistent-type-definitions */

import React, { ChangeEvent, Component, Fragment } from 'react';
import {
  EuiButton,
  EuiButtonIcon,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSuperSelect,
  EuiFieldText,
} from '@elastic/eui';
import { i18n } from '@kbn/i18n';
import { MVTFieldDescriptor, MVTFieldType } from '../../../../common/descriptor_types';
import { FieldIcon } from '../../../../../../../src/plugins/kibana_react/public';

const FIELD_TYPE_OPTIONS = [
  {
    value: MVTFieldType.String,
    inputDisplay: (
      <span>
        <FieldIcon type={'string'} />
        <span>String</span>
      </span>
    ),
  },
  {
    value: MVTFieldType.Number,
    inputDisplay: (
      <span>
        <FieldIcon type={'number'} />
        <span>Number</span>
      </span>
    ),
  },
];

export interface Props {
  fields: MVTFieldDescriptor[];
  onChange: (fields: MVTFieldDescriptor[]) => void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface State {}

export class MVTFieldConfigEditor extends Component<Props, State> {
  state = {};

  _removeField(index) {
    const newFields = this.props.fields.slice();
    newFields.splice(index, 1);

    this.props.onChange(newFields);
  }

  _addField = () => {
    const newFields = this.props.fields.slice();
    newFields.push({
      type: MVTFieldType.String,
      name: 'Foobar',
    });
    this.props.onChange(newFields);
  };

  _renderFieldTypeDropDown(mvtFieldConfig: MVTFieldDescriptor, index: number) {
    const onChange = (type: MVTFieldType) => {
      const newFields = this.props.fields.slice();
      newFields[index].type = type;
      this.props.onChange(newFields);
    };

    return (
      <EuiSuperSelect
        options={FIELD_TYPE_OPTIONS}
        valueOfSelected={mvtFieldConfig.type}
        onChange={value => onChange(value)}
      />
    );
  }

  _renderFieldNameInput(mvtFieldConfig: MVTFieldDescriptor, index: number) {
    const onChange = (e: ChangeEvent<HTMLInputElement>) => {
      const name = e.target.value;
      const newFields = this.props.fields.slice();
      newFields[index].name = name;
      this.props.onChange(newFields);
    };
    return (
      <EuiFieldText value={mvtFieldConfig.name} onChange={onChange} aria-label={'Fieldname'} />
    );
  }

  _renderFieldConfig() {
    return this.props.fields.map((mvtFieldConfig: MVTFieldDescriptor, index: number) => {
      return (
        <EuiFlexGroup key={index}>
          <EuiFlexItem>{this._renderFieldNameInput(mvtFieldConfig, index)}</EuiFlexItem>
          <EuiFlexItem>{this._renderFieldTypeDropDown(mvtFieldConfig, index)}</EuiFlexItem>
          <EuiFlexItem>
            <EuiButtonIcon
              iconType="trash"
              color="danger"
              onClick={() => {
                this._removeField(index);
              }}
              title={i18n.translate('xpack.maps.mvtSource.trashButtonTitle', {
                defaultMessage: 'Remove field',
              })}
              aria-label={i18n.translate('xpack.maps.mvtSource.trashButtonAriaLabel', {
                defaultMessage: 'Remove field',
              })}
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      );
    });
  }

  render() {
    return (
      <Fragment>
        {this._renderFieldConfig()}
        <EuiButton fill isDisabled={false} onClick={this._addField} size="s">
          {'Add field'}
        </EuiButton>
      </Fragment>
    );
  }
}
