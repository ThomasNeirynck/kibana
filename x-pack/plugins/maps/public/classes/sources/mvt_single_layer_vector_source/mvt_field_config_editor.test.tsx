/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

jest.mock('../../../kibana_services', () => ({}));

import React from 'react';
import { shallow } from 'enzyme';

import { MVTFieldConfigEditor } from './mvt_field_config_editor';
import { MVTFieldType } from '../../../../common/constants';

test('should render field editor', async () => {
  const fields = [
    {
      name: 'foo',
      type: MVTFieldType.STRING,
    },
    {
      name: 'food',
      type: MVTFieldType.STRING,
    },
    {
      name: 'fooz',
      type: MVTFieldType.NUMBER,
    },
  ];
  const component = shallow(<MVTFieldConfigEditor fields={fields} onChange={() => {}} />);

  expect(component).toMatchSnapshot();
});
