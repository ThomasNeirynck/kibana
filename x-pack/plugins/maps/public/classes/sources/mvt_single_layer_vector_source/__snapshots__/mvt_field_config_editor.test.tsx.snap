// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render error for dupes 1`] = `
<Fragment>
  <EuiFlexGroup
    gutterSize="xs"
    key="0"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={true}
        onChange={[Function]}
        placeholder="Field name"
        value="foo"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="String"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    gutterSize="xs"
    key="1"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={false}
        onChange={[Function]}
        placeholder=""
        value="bar"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="String"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    gutterSize="xs"
    key="2"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={true}
        onChange={[Function]}
        placeholder="Field name"
        value="foo"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="Number"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiSpacer
    size="xs"
  />
  <EuiButtonEmpty
    iconType="plusInCircleFilled"
    onClick={[Function]}
    size="xs"
  >
    Add
  </EuiButtonEmpty>
</Fragment>
`;

exports[`should render error for empty name 1`] = `
<Fragment>
  <EuiFlexGroup
    gutterSize="xs"
    key="0"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={false}
        onChange={[Function]}
        placeholder=""
        value="foo"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="String"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    gutterSize="xs"
    key="1"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={true}
        onChange={[Function]}
        placeholder="Field name"
        value=""
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="String"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    gutterSize="xs"
    key="2"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={false}
        onChange={[Function]}
        placeholder=""
        value="fooz"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="Number"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiSpacer
    size="xs"
  />
  <EuiButtonEmpty
    iconType="plusInCircleFilled"
    onClick={[Function]}
    size="xs"
  >
    Add
  </EuiButtonEmpty>
</Fragment>
`;

exports[`should render field editor 1`] = `
<Fragment>
  <EuiFlexGroup
    gutterSize="xs"
    key="0"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={false}
        onChange={[Function]}
        placeholder=""
        value="foo"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="String"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    gutterSize="xs"
    key="1"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={false}
        onChange={[Function]}
        placeholder=""
        value="food"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="String"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    gutterSize="xs"
    key="2"
  >
    <EuiFlexItem>
      <EuiFieldText
        aria-label="Fieldname"
        isInvalid={false}
        onChange={[Function]}
        placeholder=""
        value="fooz"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiSuperSelect
        append={
          <EuiButtonIcon
            aria-label="Remove field"
            color="danger"
            iconType="trash"
            onClick={[Function]}
            title="Remove field"
          />
        }
        compressed={false}
        fullWidth={false}
        hasDividers={false}
        isInvalid={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="string"
                />
                <span>
                  string
                </span>
              </span>,
              "value": "String",
            },
            Object {
              "inputDisplay": <span>
                <FieldIcon
                  type="number"
                />
                <span>
                  number
                </span>
              </span>,
              "value": "Number",
            },
          ]
        }
        valueOfSelected="Number"
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiSpacer
    size="xs"
  />
  <EuiButtonEmpty
    iconType="plusInCircleFilled"
    onClick={[Function]}
    size="xs"
  >
    Add
  </EuiButtonEmpty>
</Fragment>
`;
